lib/vector-core/src/event/vrl_target.rs

```diff
@@ -530,27 +524,24 @@ fn target_get_mut_metric<'a>(
 
     let value = value.get_mut(path);
 
-    for paths in path.to_alternative_components(MAX_METRIC_PATH_DEPTH) {
-        match paths.as_slice() {
-            ["name"] | ["kind"] | ["tags", _] => return Ok(value),
-            ["namespace"] | ["timestamp"] | ["tags"] => {
-                if let Some(value) = value {
-                    return Ok(Some(value));
-                }
-            }
-            _ => {
-                return Err(MetricPathError::InvalidPath {
-                    path: &path.to_string(),
-                    expected: VALID_METRIC_PATHS_SET,
-                }
-                .to_string())
-            }
+    let Some(paths) = path.to_alternative_components(MAX_METRIC_PATH_DEPTH) else {
+        return Ok(None);
+    };
+
+    match paths.as_slice() {
+        ["name"]
+        | ["kind"]
+        | ["tags", _]
+        | ["namespace"]
+        | ["timestamp"]
+        | ["interval_ms"]
```

Comment:
This was missing, feature was working without this so not sure how critical it is